src/main/kotlin/turku/forge/babyproject/resources/Resource.kt

```diff
@@ -0,0 +1,33 @@
+package turku.forge.babyproject.resources
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.annotation.JsonProperty
+import org.springframework.hateoas.LinkRelation
+import org.springframework.hateoas.RepresentationModel
+
+/**
+ * This is a `RepresentationModel` abstraction.
+ *
+ * The point of these are to be JSON representations of the values
+ * we send to the api consumer. Any class that extends this will automatically
+ * be serialized to a `hal+json` media container.
+ *
+ * since we don't yet know what media container we will use
+ * this class serves as a abstraction for the media container we are currently using.
+ *
+ * Its likely that this will be removed once we have chosen a media container
+ * but we may keep it who knows
+ */
+abstract class Resource<T : Resource<T>> : RepresentationModel<T>() {
+    @JsonProperty("_embedded")
```

Comment:
Might be worth looking at `HalModelBuilder` to mange embedding in general.

Reply:
There is actualy a good reason for not using it. Or ok there is a reason. We don't really know if we want to use HAL or something else. We are in the prosses figuring that out. I figured this would be a eayse and simple way of hiding away what is beeing used. Because if we don't use HAL we need to have some other type of Modle bulder for what ever else we end up use. But yea I totlay could have used that for the one request that returns the HAL